bitwarden_license/src/Commercial.Core/SecretManagerFeatures/Secrets/UpdateSecretCommand.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Bit.Commercial.Core.SecretManagerFeatures.Secrets.Interfaces;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+
+namespace Bit.Commercial.Core.SecretManagerFeatures.Secrets
+{
+    public class UpdateSecretCommand : IUpdateSecretCommand
+    {
+        private readonly ISecretRepository _secretRepository;
+
+        public UpdateSecretCommand(ISecretRepository secretRepository)
+        {
+            _secretRepository = secretRepository;
+        }
+
+        public async Task<Secret> UpdateAsync(Secret secret)
+        {
+            if (secret.Id == default(Guid))
+            {
+                throw new Exception("Cannot update secret, secret does not exist.");
```

Comment:
We might want to use a `BadRequestException` here. If it were to get bubbled up to the controller it would result in a status 500 error.

Reply:
resolved in https://github.com/bitwarden/server/pull/2201/commits/8104854ca81795a40cfd44c623d23746d2fccf99